### PR TITLE
DRILL-5961: For long running queries (> 10 min) Drill may raise FragmentSetupException for completed/cancelled fragments

### PR DESCRIPTION
**Pull request by vrozov**

### CODE REVIEW

**Comment by priteshm** [comment]
@parthchandra or @adityakishore since you'll have made changes to these files before, can you review the changes here?

File: exec/java-exec/src/main/java/org/apache/drill/exec/work/WorkManager.java [hunk 1]
```diff
@@ -277,7 +277,9 @@ public void startFragmentPendingRemote(final FragmentManager fragmentManager) {
         @Override
         protected void cleanup() {
           runningFragments.remove(fragmentHandle);
-          workBus.removeFragmentManager(fragmentHandle);
+          if (!fragmentManager.isCancelled()) {
+            workBus.removeFragmentManager(fragmentHandle, false);
```
Comment (parthchandra) [inline exec/java-exec/src/main/java/org/apache/drill/exec/work/WorkManager.java]: Not sure why you don't want to cancel here.

File: exec/java-exec/src/main/java/org/apache/drill/exec/rpc/control/WorkEventBus.java [hunk 2]
```diff
@@ -74,83 +70,42 @@ public void statusUpdate(final FragmentStatus status) {
...
       }
-
-      manager.cancel();
-      recentlyFinishedFragments.put(handle, 1);
-      managers.remove(handle);
       return true;
+    } else if (logger.isWarnEnabled()) {
+      logger.warn("Fragment {} manager is not found in the work bus.", QueryIdHelper.getQueryIdentifier(handle));
     }
+    return false;
```
Comment (parthchandra) [inline exec/java-exec/src/main/java/org/apache/drill/exec/rpc/control/WorkEventBus.java]: Seems like this is redundant. Doesn't seem like we ever return false, and the callers never seem to check.

File: exec/java-exec/src/main/java/org/apache/drill/exec/work/WorkManager.java (hunk 1 above)
Comment (vrozov) [reply exec/java-exec/src/main/java/org/apache/drill/exec/work/WorkManager.java]: If cleanup is not called as a result of FragmentManager cancellation, it is part of the regular cleanup after run is complete.

File: exec/java-exec/src/main/java/org/apache/drill/exec/rpc/control/WorkEventBus.java (hunk 2 above)
Comment (vrozov) [reply exec/java-exec/src/main/java/org/apache/drill/exec/rpc/control/WorkEventBus.java]: ControlMessageHandler checks the result on line 196.

**Pull request closed without merging**